SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1423,202 +1393,140 @@ bool RunSchemaCompiler(FString& SchemaBundleJsonOutput, FString SchemaInputDir,
-				OutComponentIdToFieldIdsIndex.Add(ComponentId, OutFieldIdsArray.Num());
-				TArray<uint32>& FieldIDs = OutFieldIdsArray.AddDefaulted_GetRef().FieldIds;
-
-				for (const auto& ArrayValue : *FieldArray)
-				{
-					const TSharedPtr<FJsonObject>* ArrayObject;
-					SAFE_TRYGET(ArrayValue, Object, ArrayObject);
-
-					int32 FieldId;
-					SAFE_TRYGETFIELD((*ArrayObject), Number, "fieldId", FieldId);
-
-					COND_SCHEMA_GEN_ERROR_AND_RETURN(FieldIDs.Contains(FieldId),
-													 TEXT("The schema bundle contains duplicate fieldId: %d, component name: %s."), FieldId,
-													 *ComponentName);
-					FieldIDs.Add(FieldId);
-				}
-			}
+		return true;
+	});
 
-			FString DataDefinition;
-			SAFE_TRYGETFIELD((*CompObject), String, "dataDefinition", DataDefinition);
+	FString SchemaJsonPath;
+	if (!RunSchemaCompiler(SchemaJsonPath, SchemaInputPath))
+	{
+		UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Failed to parse partition meta data schema files"));
+		return false;
+	}
 
-			if (!DataDefinition.IsEmpty())
-			{
-				COND_SCHEMA_GEN_ERROR_AND_RETURN(
-					FieldArray->Num() != 0,
-					TEXT("The schema bundle supplied both a data definition and field IDs - this is unexpected, component name: %s."),
-					*ComponentName);
-				ComponentIdToDataDefinitionName.Add(ComponentId, DataDefinition);
-			}
-		}
+	TSet<FString> SchemaFileNames;
+	for (const FString& FilePath : SchemaFiles)
+	{
+		SchemaFileNames.Add(FPaths::GetCleanFilename(FilePath));
+	}
 
-		const TArray<TSharedPtr<FJsonValue>>* ComponentSets;
-		SAFE_TRYGETFIELD((*FileObject), Array, "componentSets", ComponentSets);
+	TArray<SpatialGDK::SchemaComponentIdentifiers> Components;
+	SpatialGDK::ExtractComponentsFromSchemaJson(SchemaJsonPath, Components, SchemaFileNames);
 
-		for (const auto& CompSetValue : *ComponentSets)
+	FString DestinationSchemaDir = FPaths::Combine(SchemaOutputPath, PartitionDataFolderName);
+	if (FPaths::DirectoryExists(DestinationSchemaDir))
+	{
+		if (!Filesystem.DeleteDirectoryRecursively(*DestinationSchemaDir))
 		{
-			const TSharedPtr<FJsonObject>* CompSetObject;
-			SAFE_TRYGET(CompSetValue, Object, CompSetObject);
-
-			int32 ComponentSetId;
-			SAFE_TRYGETFIELD((*CompSetObject), Number, "componentSetId", ComponentSetId);
-
-			const TSharedPtr<FJsonObject>* CompListObject;
-			SAFE_TRYGETFIELD((*CompSetObject), Object, "componentList", CompListObject);
-
-			const TArray<TSharedPtr<FJsonValue>>* RefComponents;
-			SAFE_TRYGETFIELD((*CompListObject), Array, "components", RefComponents);
-
-			TSet<FString> Components;
-
-			for (const auto& CompRefValue : *RefComponents)
-			{
-				const TSharedPtr<FJsonObject>* CompRefObject;
-				SAFE_TRYGET(CompRefValue, Object, CompRefObject);
-
-				FString ComponentName;
-				SAFE_TRYGETFIELD((*CompRefObject), String, "component", ComponentName);
-
-				Components.Add(ComponentName);
-			}
-
-			ComponentRefSetMap.Add(ComponentSetId, MoveTemp(Components));
+			// clang-format off
```

Comment:
So there was a very strange thing that happened when letting clang-format run on this file and format the log directives.
The lines in the PDB did not match the source....
So you would have a breakpoint and it would hit 20 lines above.
Drove me crazy until I stopped clang-format from breaking up the code in macros. Then the PDB info matched the sources again.

Reply:
Worth a comment I would think.

Reply:
There is a comment line 509.
There are 30 tags throughout the file.
Should each tag get its own comment?